source/Command.cpp

```diff
@@ -274,29 +291,30 @@ double Command::Turn() const
 // Check if this command includes a command to fire the given weapon.
 bool Command::HasFire(int index) const
 {
-	if(index < 0 || index >= 32)
+	if(index < 0 || index >= 512)
 		return false;
 	
-	return state & ((1ull << 32) << index);
+	return firing_weps[index >> 6] & (1ull << (index % 64));
 }
 
 
 
 // Add to this set of commands a command to fire the given weapon.
 void Command::SetFire(int index)
 {
-	if(index < 0 || index >= 32)
+	if(index < 0 || index >= 512)
 		return;
 	
-	state |= ((1ull << 32) << index);
+	firing_weps[index >> 6] |= (1ull << (index % 64));
 }
 
 
 
 // Check if any weapons are firing.
 bool Command::IsFiring() const
 {
-	return (state & 0xFFFFFFFF00000000ull);
+	return (bool) (firing_weps[0] | firing_weps[1] | firing_weps[2] | firing_weps[3] |
```

Comment:
C-style casts should be avoided whenever possible. If you must cast explicitly, use `static_cast`

Reply:
Switched to using logical or instead of bitwise or to reap possible performance gains from short-circuit logic that bitwise or ignores.